pkg/cluster/self_hosted.go

```diff
@@ -43,24 +43,32 @@ func (c *Cluster) addOneSelfHostedMember() error {
 	}
 	// wait for the new pod to start and add itself into the etcd cluster.
 	oldN := c.members.Size()
-	err = retryutil.Retry(5*time.Second, math.MaxInt64, func() (bool, error) {
+	err = retryutil.Retry(10*time.Second, 6, func() (bool, error) {
 		err = c.updateMembers(c.members)
 		if err != nil {
-			c.logger.Errorf("add self hosted member: fail to update members: %v", err)
```

Comment:
This change is necessary.
The previous logic is somewhat problematic. It could exit the loop silently without waiting the new member added.
We should just retry on any failure.